cardinal/ecs/messages/manager.go

```diff
@@ -0,0 +1,110 @@
+package msgs
+
+import (
+	"errors"
+	"github.com/rotisserie/eris"
+	"pkg.world.dev/world-engine/cardinal/types/message"
+	"slices"
+)
+
+var ErrDuplicateMessageName = errors.New("message names must be unique")
+
+type Manager struct {
+	registeredMessages map[string]message.Message
+	nextMessageID      message.TypeID
+}
+
+func New() *Manager {
+	return &Manager{
+		registeredMessages: map[string]message.Message{},
+		nextMessageID:      1,
+	}
+}
+
+// RegisterMessages registers the list of message.Messages with the Manager. Returns an error if any of the
+// messages have duplicate names.
+func (m *Manager) RegisterMessages(msgs ...message.Message) error {
+	// We check for duplicate message names within the slice and against the map of registered messages.
+	// This ensures that we are registering all messages in the slice, or none of them.
+	var seenNames []string
+	for _, msg := range msgs {
+		// Check for duplicate message names against within the slice
+		if slices.Contains(seenNames, msg.Name()) {
```

Comment:
no reason in particular, used slice because `msgs` is already a slice. non-intuitively i think it's also faster at small number of messages because you can preallocate it and it's contiguous memory

Reply:
also it's more 1:1 to the system manager to use slice (and the system manager specifically wants a slice because it cares about ordering, which maps are not able to provide due to non deterministic key iter)